File: Codeforces/Round_328/C.rb
t, w, b = gets.strip.split.map(&:to_i)

l = w.lcm(b)
a = t/l * [w, b].min
a += [t % l + 1, [w, b].min].min
a -= 1

if w == b then
  puts "1/1"
else
  g = a.gcd(t)
  puts "#{a/g}/#{t/g}"
end

File: Codeforces/Round_328/C.cpp
#include <iostream>
#include <algorithm>

using namespace std;

long long gcd(long long a, long long b) {
  if(b == 0) return a;
  else return gcd(b, a % b);
}

long long lcm(long long a, long long b) {
  return (a * b)/gcd(a, b);
}

int main() {
  long long t, w, b; cin >> t >> w >> b;

  long long a = t/lcm(w, b) * min(w, b);
  a += min(min(w, b) - 1, t);

  cout << a/gcd(a, t) << "/" << t/gcd(a, t) << endl;
}
